scripts/get_collection_ids.cdc

```diff
@@ -3,13 +3,12 @@
 import "NonFungibleToken"
 import "ExampleNFT"
 
-pub fun main(address: Address, collectionPublicPath: PublicPath): [UInt64] {
+access(all) fun main(address: Address, collectionPublicPath: PublicPath): [UInt64] {
     let account = getAccount(address)
 
-    let collectionRef = account
-        .getCapability(collectionPublicPath)
-        .borrow<&{NonFungibleToken.CollectionPublic}>()
-        ?? panic("Could not borrow capability from public collection at specified path")
+    let collectionRef = account.capabilities.borrow<&{NonFungibleToken.Collection}>(
```

Comment:
When testing migration, I had to change the above borrow to this:
```
account.capabilities.borrow<&{
        NonFungibleToken.CollectionPublic,
        ExampleNFT.ExampleNFTCollectionPublic,
        ViewResolver.ResolverCollection
    }>(
        /public/exampleNFTCollection
      ) ?? panic("Could not borrow capability from collection at specified path")
```
Is this something only after migration?

Reply:
@SupunS do we maybe need to add the same special case here as you did for interface conformance so that we can use the `&{NonFungibleToken.Collection}` restricted type? 👀

Reply:
This is actually not related to the contract update restrictions (https://cadence-lang.org/docs/1.0/language/contract-updatability), to which the special casing was done. Those restrictions are applicable to field types and conformances only, just to prevent existing data from getting corrupted.

The type here, can be changed freely. However, the catch is, the stored value in that path also gets migrated during the storage migration (There is some detailed explanation [here](https://github.com/onflow/flips/blob/main/cadence/20230601-entitlements-migration.md) as to why the values are migrated this way). So even if you use the old type, when the script is executed, it will complain saying the stored value doesn't match the requested type.

_Just for some additional context: The migration has two steps: 1) Updating the staged contracts / contract code. 2) Migrating existing data. This particular problem is related to the step (2), where as, the type-validations/contract-update-restrictions are applicable only to the step (1)._

Reply:
It sounds like with the migration process, there would some differences with how stored values can be accessed that might not apply to contracts starting from 1.0 as shown in this example.
Is there a paragraph/doc in the migration guide capturing this information?

Reply:
>  It sounds like with the migration process, there would some differences with how stored values can be accessed

Yes. Don't think we have any documentation around that yet. Good idea to add a section explaining this. Could you please create an issue in cadence repo?

Reply:
Good idea, just opened https://github.com/onflow/cadence/issues/3206
Feel free to edit or add any comments for clarification.

Reply:
shouldn't it be:
```cadence 
account.capabilities.borrow<&{
        NonFungibleToken.CollectionPublic
    }>(
        /public/exampleNFTCollection
      ) ?? panic("Could not borrow capability from collection at specified path")
```

Reply:
I actually was wondering the same. @dsainati1 do you know whether this is expected?

Reply:
ah ok, so it seems the original capability was linked with all thee interfaces: https://github.com/onflow/flow-nft/blob/4c0c684baed98e2ffd7a8f9d27973c1db5dd1239/contracts/ExampleNFT.cdc#L398

So when the link gets converted to a capability, it retains the same intersection type.

Reply:
That might be because the way it is linked before the migration doesn't get migrated cleanly. Because `{NonFungibleToken.Collection}` didn't exist before, I guess that the migration doesn't pick up that is is now possible to borrow as that type. The way I changed it to here is the way that I would prefer people do it in the future though. Not sure if that is more or less important than making it clearer for migrations

Reply:
@SupunS 
> So when the link gets converted to a capability, it retains the same set of interfaces.
but doesn't it mean any one of them would work?

Reply:
ah yes, it should (and does) work with `&{NonFungibleToken.CollectionPublic}`. I got it confused with the `NonFungibleToken.Collection` and didn't realize these two are two different interfaces (thought one got replaced with the other, but they both exist)